backend/modules/simulation/router.py

```
@@ -235,6 +235,66 @@ async def save_message(
         raise HTTPException(status_code=500, detail="Internal server error")
 
 
+@router.post("/execute-code", response_model=CodeExecutionResponse)
+async def execute_code(
+    request: CodeExecutionRequest,
+    current_user: User = Depends(get_current_user),
```

Ruff (0.15.0)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)